public/app/features/alerting/unified/hooks/useCombinedRule.ts

```diff
@@ -116,6 +118,12 @@ export function useCombinedRule({ ruleIdentifier, limitAlerts }: Props): Request
       refetchOnMountOrArgChange: true,
     }
   );
+  // in case of Grafana folder, we need to use the folder name instead of uid, as in promrules we don't use uid
+  const folder = useFolder(ruleLocation?.namespace);
+  const folderName =
+    ruleSourceName === GRAFANA_RULES_SOURCE_NAME && folder.folder
+      ? stringifyFolder(folder.folder)
+      : ruleLocation?.namespace;
```

Comment:
I think we can simplify this a bit
```suggestion
	const isGrafanaRule = isGrafanaRulesSource(ruleSourceName);
  const folder = useFolder(isGrafanaRule ? ruleLocation?.namespace : undefined);
  const namespaceName =  isGrafanaRule && folder.folder
      ? stringifyFolder(folder.folder)
      : ruleLocation?.namespace;
```